### PR TITLE
The memory freeing problem resolved

### PR DESCRIPTION
**Pull request by NickPepper**
plus a little code cleanup for better readability

### CODE REVIEW

File: examples/hello_world.js [hunk 1]
```diff
@@ -69,14 +69,13 @@ function main() {
   }
 
   // Delete objects we created through |new|. We just do a few of them here, but you should do them all if you are not shutting down ammo.js
-  Ammo.destroy(collisionConfiguration);
-  Ammo.destroy(dispatcher);
-  Ammo.destroy(overlappingPairCache);
+  Ammo.destroy(dynamicsWorld);
```
Comment (kripken) [inline examples/hello_world.js]: was the issue that the order needs to be like this? if so then perhaps add a comment here?

Comment (NickPepper) [inline examples/hello_world.js]: sure we always need to call free() in reversed order against our malloc() it is so clear that I'm not sure the special comment needed

Comment (kripken) [inline examples/hello_world.js]: why? in general the order of mallocs does not need to be the same as free. except here, perhaps the destructor of the main object assumes its children still exist. so there is a reason here, I think, but it's specific to these objects, not to malloc/free in general. so a comment might help people see that.

Comment (NickPepper) [inline examples/hello_world.js]: done :) btw nice to meet you and this is a great project, really

Comment (kripken) [inline examples/hello_world.js]: Great, thanks! :)

**Pull request merged**